File: model/group_style.go
package model

// GroupStyle 团队共享的自定义风格（图片 / 视频）。
//
// 与「我的风格」的关系：个人风格存在用户自己的 sync_data "presets" 域里（私有、跨设备同步）；
// 分享到团队是【复制】一份到这张表，个人那份保留。所以取消共享不会连带删掉本人的风格。
//
// 预览图为什么要另存一份、而不是直接引用创建者的 previewStorageKey：
// serveSyncFile 的跨账号回退（handler/sync.go:507）只对公共读桶的文件生效，
// 落到本地磁盘的文件仍按账号隔离；实际部署中会有相当一部分文件落在磁盘上，直接引用会让组内其他人看到空白缩略图。
// 所以分享时把预览图字节一并复制进 group-styles/<groupId>/，与 GroupAsset 同款做法。
type GroupStyle struct {
	ID          string `json:"id" gorm:"primaryKey"`
	GroupID     string `json:"groupId" gorm:"index"`
	OwnerUserID string `json:"ownerUserId" gorm:"index"`
	OwnerName   string `json:"ownerName"` // 分享者展示名（displayName 优先，回退 username）
	Kind        string `json:"kind"`      // image | video

	NameZh         string `json:"nameZh"`
	Description    string `json:"description"`
	PrefixPrompt   string `json:"prefixPrompt"` // 仅视频风格用：放在用户描述【之前】
	InjectPrompt   string `json:"injectPrompt"`
	NegativePrompt string `json:"negativePrompt"`

	PreviewURL      string `json:"previewUrl"` // TOS 公网 URL；落盘时为磁盘路径（列表时换成 content 端点）
	PreviewMimeType string `json:"previewMimeType"`

	// SourceStyleID 记录它是从哪个个人风格分享来的，用于「已分享」状态判断与去重。
	SourceStyleID string `json:"sourceStyleId" gorm:"index"`

	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
}

File: web/src/services/api/group-style.ts
import axios from "axios";

import { apiGet, apiDelete } from "@/services/api/request";
import { useUserStore } from "@/stores/use-user-store";

// 团队共享风格：同组成员都能选用；分享者本人或管理员可取消共享。
// 与「我的风格」的区别：后者存在个人 presets 云同步域里（私有），这里走服务端表、按 group_id 可见。

export type GroupStyleKind = "image" | "video";

export type GroupStyle = {
    id: string;
    groupId: string;
    ownerUserId: string;
    ownerName: string;
    kind: GroupStyleKind;
    nameZh: string;
    description: string;
    prefixPrompt: string;
    injectPrompt: string;
    negativePrompt: string;
    previewUrl: string;
    previewMimeType: string;
    sourceStyleId: string;
    createdAt: string;
    updatedAt: string;
};

function token() {
    const value = useUserStore.getState().token;
    if (!value) throw new Error("请先登录");
    return value;
}

export async function listGroupStyles() {
    return apiGet<GroupStyle[]>("/api/v1/group-styles", undefined, token());
}

export type ShareGroupStyleInput = {
    kind: GroupStyleKind;
    sourceStyleId: string;
    nameZh: string;
    description?: string;
    prefixPrompt?: string;
    injectPrompt: string;
    negativePrompt?: string;
    /** 预览图字节。必须一并上传：跨账号只回退公共读桶，落在磁盘上的预览图组内其他人取不到。 */
    preview?: Blob;
};

export async function shareGroupStyle(input: ShareGroupStyleInput) {
    const body = new FormData();
    body.append("kind", input.kind);
    body.append("source_style_id", input.sourceStyleId);
    body.append("name_zh", input.nameZh);
    body.append("inject_prompt", input.injectPrompt);
    if (input.description) body.append("description", input.description);
    if (input.prefixPrompt) body.append("prefix_prompt", input.prefixPrompt);
    if (input.negativePrompt) body.append("negative_prompt", input.negativePrompt);
    if (input.preview) body.append("preview", input.preview, "preview.png");
    const response = await axios.post("/api/v1/group-styles", body, { headers: { Authorization: `Bearer ${token()}` }, validateStatus: () => true });
    const payload = response.data as { code?: number; data?: GroupStyle; msg?: string };
    if (!payload || payload.code !== 0 || !payload.data) throw new Error(payload?.msg || "分享到团队失败");
    return payload.data;
}

export async function deleteGroupStyle(id: string) {
    return apiDelete<{ ok: boolean }>(`/api/v1/group-styles/${encodeURIComponent(id)}`, token());
}
